torch/_inductor/codegen/triton.py

```diff
@@ -1948,7 +1957,7 @@ def load(self, name: str, index: sympy.Expr):
         if advance_block_ptr:
             load_buffer.writeline(advance_block_ptr)
 
-        if not self.inside_reduction or not has_rindex:
+        if not self.inside_reduction or (not indexing.has_rmask() and not has_rindex):
```

Comment:
shouldn't this condition be 
```
if not self.inside_reduction or load_buffer == self.body:
```
? That is, a variable is outside a loop if there was no loop to start with or we were able to lift it to he body of the function.